lib/widgets/image.dart

```diff
@@ -149,3 +151,59 @@ enum ImageAnimationMode {
     }
   }
 }
+
+extension ImageThumbnailLocatorExtension on ImageThumbnailLocator {
+  /// Chooses an appropriate format from [PerAccountStore.serverThumbnailFormats],
+  /// represented as an absolute URL.
+  ///
+  /// [height] and [width] are in logical pixels.
+  ///
+  /// Requires an ancestor [PerAccountStoreWidget].
+  Uri? resolve(
+    BuildContext context, {
+    required double width,
+    required double height,
+    required ImageAnimationMode animationMode,
+  }) {
+    final devicePixelRatio = MediaQuery.devicePixelRatioOf(context);
+    final widthPhysicalPx = (width * devicePixelRatio).ceil();
+    final heightPhysicalPx = (height * devicePixelRatio).ceil();
+
+    final store = PerAccountStoreWidget.of(context);
+    ThumbnailFormat? bestCandidate;
+
+    final animateIfSupported = animationMode.resolve(context);
+    if (hasAnimatedVersion && animateIfSupported) {
+      bestCandidate ??= _bestFormatOf(store.sortedAnimatedThumbnailFormats,
+                          width: widthPhysicalPx, height: heightPhysicalPx);
+    }
+
+    bestCandidate ??= _bestFormatOf(store.sortedStillThumbnailFormats,
+                        width: widthPhysicalPx, height: heightPhysicalPx);
```

Comment:
nit: join up the bestCandidate computation as one stanza (and separate it from getting `store`)